slowquant/qiskit_interface/interface.py

```diff
@@ -160,70 +176,124 @@ def construct_circuit(self, num_orbs: int, num_elec: tuple[int, int]) -> None:
                     f"Got parameter name, {name}, that is not in grad_param_R, {self.grad_param_R}"
                 )
 
-        self.num_qubits = self.circuit.num_qubits
-        # Set parameter to HarteeFock
-        self._parameters = [0.0] * self.circuit.num_parameters
+        if not reconstruct:
```

Comment:
So I need it not to reset for change_primitive in wavefunction. 
You wrote this initial line of resetting parameters :P I assume you use it in VQE?

Reply:
It is not resetting parameters xd
It is creating the attribute _parameters

So I guess instead of the `reconstruct`, doing,

```
        if not hasattr(self, '_parameters'):
            self._parameters = [0.0] * self.circuit.num_parameters
```

would do the same thing

Reply:
lol. yes. I read that line too quick. will change accordingly.